runhouse/rns/hardware/cluster.py

```diff
@@ -542,6 +556,29 @@ def run(
                 env = Env.from_name(env)
             cmd_prefix = env._run_cmd
 
+        if name_run:
```

Comment:
I'm kinda confused about all the different methods to run on the cluster. If the user doesn't pass name_run, the run happens through the runner like it does today, but if they do it goes over rpc?

Reply:
we need a way to basically intercept the command before running it in order to create the Run object + config,  stdout, etc. on the cluster, so that's why we use an RPC here. Do you think we should always use an RPC instead of splitting the functionality like this?